runparams.py

```diff
@@ -3,87 +3,148 @@
+        if self.generate_mkdocs:
+            check_import_module("jinja2", "MkDocs generation")
+        if self.generate_rdf:
+            check_import_module("rdflib", "RDF generation")
+        if self.generate_tex:
+            check_external_program("pandoc", "TeX generation")
+            check_import_module("jinja2", "TeX generation")
+
+    def process_args(self, opts=sys.argv[1:]):
+        def check_input_path(p):
+            if not p.exists():
+                raise argparse.ArgumentTypeError(f"Input directory '{p}' does not exist.")
+            if not p.is_dir():
+                raise argparse.ArgumentTypeError(f"Input path '{p}' is not a directory.")
+            if p.name != "model":
+                raise argparse.ArgumentTypeError(f"Input directory '{p}' must be named 'model'.")
+
+        parser = argparse.ArgumentParser(description="Generate documentation from an SPDXv3 model.")
+
+        parser.add_argument("input_dir", type=str, help="Path to the input 'model' directory.")
+
         parser.add_argument("-d", "--debug", action="store_true", help="Print debug output")
-        parser.add_argument("-f", "--force", action="store_true", help="Overwrite existing generated files")
-        parser.add_argument("-n", "--nooutput", action="store_true", help="Do not generate anything, only check input")
-        parser.add_argument("-q", "--quiet", action="store_true", help="Print no output")
+        parser.add_argument("-f", "--force", action="store_true", help="Force overwrite of existing output directories.")
+        parser.add_argument("-j", "--generate-jsondump", action="store_true", help="Generate a dump of the model in JSON format.")
+        parser.add_argument("-J", "--output-jsondump", type=str, help="Output directory for JSON dump file.")
+        parser.add_argument("-m", "--generate-mkdocs", action="store_true", help="Generate mkdocs output.")
+        parser.add_argument("-M", "--output-mkdocs", type=str, help="Output directory for mkdocs files.")
+        parser.add_argument("-n", "--no-output", action="store_true", help="Perform no output generation, only input validation.")
+        parser.add_argument("-o", "--output", type=str, help="Single output directory for all output types.")
+        parser.add_argument("-p", "--generate-plantuml", action="store_true", help="Generate PlantUML output.")
+        parser.add_argument("-P", "--output-plantuml", type=str, help="Output directory for PlantUML files.")
+        parser.add_argument("-r", "--generate-rdf", action="store_true", help="Generate RDF output.")
+        parser.add_argument("-R", "--output-rdf", type=str, help="Output directory for RDF files.")
+        parser.add_argument("-t", "--generate-tex", action="store_true", help="Generate TeX output.")
+        parser.add_argument("-T", "--output-tex", type=str, help="Output directory for TeX files.")
         parser.add_argument("-v", "--verbose", action="store_true", help="Print verbose output")
```

Comment:
```suggestion
        parser.add_argument("-v", "--verbose", action="store_true", help="Print verbose output.")
```

Other descriptions have a fullstop.